modules/virtualization/microvm/appvm.nix

```diff
@@ -0,0 +1,176 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  configHost = config;
+  cfg = config.ghaf.virtualization.microvm.appvm;
+  waypipe-ssh = pkgs.callPackage ../../../user-apps/waypipe-ssh {};
+
+  makeVm = {vm}: let
+    hostname = "vm-" + vm.name;
```

Comment:
For consistency with system VMs (`netvm`, `guivm`, `adminvm`) we could also consider postfixing (`-vm`) appVMs instead of prefixing (`vm-`).

Reply:
Either way, let's keep '-' sign somewhere. For my taste, 'someappnamevm' or 'vmsomeappname' looks not so obvious, than 'someappname-vm' or 'vm-someappname'

Reply:
Will be a part of SP-2839 task